pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -203,14 +204,14 @@ func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
 			}
 			cfg[url] = *cred
 		}
+		// add ACR anonymous repo support: use empty username and password for anonymous access
+		cfg["*.azurecr.*"] = credentialprovider.DockerConfigEntry{
+			Username: "",
+			Password: "",
+			Email:    dummyRegistryEmail,
+		}
 	}
 
-	// add ACR anonymous repo support: use empty username and password for anonymous access
```

Comment:
why remove this? when user use `UseManagedIdentityExtension`,  also needs to support ACR anonymous.
Original code will always have `empty username and password for anonymous access`, will that solve your issue?

Reply:
The provided credential may not be authz to the ACR, so empty auth context is required for both MSI and SPN.